provider/machinepool/machine_pool_resource.go

```diff
@@ -544,14 +544,9 @@ func (r *MachinePoolResource) Update(ctx context.Context, req resource.UpdateReq
 
 	object := update.Body()
 
-	// update the autoscaling enabled with the plan value (important for nil and false cases)
-	state.AutoScalingEnabled = plan.AutoScalingEnabled
```

Comment:
While AutoScalingEnabled can be not computed, I think Replicas need to be computed because when user doesn't declare value we put default =2
Anyway this should be done in populate and not here.

@JohnStrunk @bardielle @sagidayan what do you think?

I hope we will have here one convention how those cases should be handled